includes/prometheus.inc.php

```diff
@@ -26,19 +26,28 @@ function prometheus_push($device, $measurement, $tags, $fields)
 
                 foreach ($fields as $k => $v) {
                     if ($v !== null) {
+                        $k = preg_replace('/([a-z]{1,})/', '$1_', $k); // Generally, metric naming in Prometheus is done with underscores without capital letters
```

Comment:
You could just use this function...

Reply:
Snake case can be good, but unfortunately an array of metrics come with all capital letters like INBROADCASTPKTS, INDISCARDS, INERRORS, INMULTICASTPKTS, etc., which are then translated to i_n_b_r_o_a_d_c_a_s_t_p_k_t_s, i_n_d_i_s_c_a_r_d_s, etc. This is obviously unreadable, but this is how data is received by "prometheus_push()". If I understood correctly, it comes from includes/datastore.inc.php via data_update function, and it is called therefore from an array of places... How should I proceed?
Here is a poller debug, just in case: https://gist.github.com/angryp/c601f9cf4869b470c2e10a4affc68616

Reply:
Snake case alone won't do the job ;-)